src/nvm_x722.cpp

```diff
@@ -167,3 +171,161 @@ NvmX722::word_t NvmX722::calcChecksum() const
 
     return checksumBase - chkSum;
 }
+
+/**
+ * @brief Convert hex character to binary value
+ */
+inline uint8_t charToByte(const char c)
+{
+    if (c >= '0' && c <= '9')
+    {
+        return static_cast<uint8_t>(c - '0');
+    }
+    else if (c >= 'a' && c <= 'f')
+    {
+        return static_cast<uint8_t>(10 + c - 'a');
+    }
+    else if (c >= 'A' && c <= 'F')
+    {
+        return static_cast<uint8_t>(10 + c - 'A');
+    }
+
+    throw std::invalid_argument("Illegal character");
+}
+
+struct BoardInfo
+{
+    static constexpr uint8_t macAddressType = 0x01;
+    static constexpr uint8_t x722MacAddr = 0x03;
+
+    uint8_t recordType = 0;
+    uint8_t addrType = 0;
+    NvmX722::mac_t mac = {0};
+
+    /**
+     * @brief Get record designation.
+     */
+    inline uint8_t designation() const
+    {
+        return static_cast<uint8_t>((addrType & 0xF0) >> 4);
+    }
+
+    /**
+     * @brief MAC address index
+     */
+    inline uint8_t index() const
+    {
+        return static_cast<uint8_t>(addrType & 0x0F);
+    }
+
+    /**
+     * @brief Parse hex string into BoardInfo structure
+     *
+     * @param s - Hex string
+     *
+     * @return BoardInfo structure
+     *
+     * @throw std::exception in case of errors
+     */
+    static BoardInfo fromString(const std::string& s)
+    {
+        BoardInfo bi;
+        uint8_t* bytes = reinterpret_cast<uint8_t*>(&bi);
+        const size_t len = std::min(s.length(), sizeof(bi) * 2);
+        for (size_t i = 0; i < len; ++i)
+        {
+            size_t idx = i / 2;
+            if ((i % 2) == 0)
+            {
+                bytes[idx] = charToByte(s[i]) << 4;
+            }
+            else
+            {
+                bytes[idx] |= charToByte(s[i]);
+            }
+        }
+
+        return bi;
+    }
+} __attribute__((packed));
+
+using Property = std::variant<std::string>;
+using Properties = std::map<PropertyName, Property>;
+/**
+ * @brief Get all D-Bus properties from specifice service.
+ *
+ * NOTE: this not in dbus.hpp as we expects the string only properties.
```

Comment:
`as we expect only the string properties` ? Anyway, I can't see how this justifies this not being in `dbus.hpp`.

Reply:
fixed.